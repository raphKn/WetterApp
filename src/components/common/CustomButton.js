import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, TouchableOpacity, ViewPropTypes } from 'react-native';
import CustomText from './CustomText';

const styles = StyleSheet.create({
    container: {
        margin: 20,
        backgroundColor: '#03A9F4',
        height: 32,
        justifyContent: 'center',
        alignItems: 'center',
    },
    buttonText: { paddingHorizontal: 16 }
});

const CustomButton = props => (<TouchableOpacity style={[styles.container, props.style]} onPress={props.onPress}>
    <CustomText style={styles.buttonText}>{props.text.toLocaleUpperCase()}</CustomText>
</TouchableOpacity>);

CustomButton.propTypes = {
    text: PropTypes.string.isRequired,
    onPress: PropTypes.func.isRequired,
    style: ViewPropTypes.style,
};

CustomButton.defaultProps = {
    style: null,
};

export default CustomButton;
