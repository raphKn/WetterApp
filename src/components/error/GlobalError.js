import React from 'react';
import PropTypes from 'prop-types';
import { Dimensions, StyleSheet, View } from 'react-native';
import CustomText from '../common/CustomText';
import CustomButton from '../common/CustomButton';

const styles = StyleSheet.create({
    container: {
        position: 'absolute',
        left: 0,
        top: 0,
        backgroundColor: 'rgba(114,114,114,0.6)',
        justifyContent: 'center',
        alignItems: 'center',
        padding: 20,
    },
    innerContainer: {
        backgroundColor: 'white',
        alignItems: 'center',
        paddingHorizontal: 12,
        paddingTop: 12,
        borderRadius: 10,

    },
    title: {
        fontSize: 20,
        fontWeight: 'bold',
        marginBottom: 20,
    },
    message: {
        textAlign: 'center',
    },
});

const GlobalError = (props) => {
    if (!props.message) {
        return null;
    }
    const { width, height } = Dimensions.get('window');
    return (

        <View style={[styles.container, { height, width }]}>
            <View style={styles.innerContainer}>
                <CustomText style={styles.title}>Es ist ein Fehler aufgetreten</CustomText>
                <CustomText style={styles.message}>{props.message}</CustomText>
                <CustomButton text="ZurÃ¼ck" onPress={props.cancelError} />
            </View>
        </View>
    );
}

GlobalError.propTypes = {
    message: PropTypes.string,
    cancelError: PropTypes.func,
};

GlobalError.defaultProps = {
    message: '',
    cancelError: () => {
    },
};

export default GlobalError;
