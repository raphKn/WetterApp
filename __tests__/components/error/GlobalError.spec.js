import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import GlobalError from '../../../src/components/error/GlobalError';
import CustomButton from '../../../src/components/common/CustomButton';

describe('GlobalError', () => {
    it('renders  correctly in a snapshot', () => {
        const doNothing = () => {
        };
        const wrapper = renderer.create(
            <GlobalError message="Fahlermessage" cancelError={doNothing} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders no error for empty message', () => {
        const doNothing = () => {
        };
        const wrapper = renderer.create(
            <GlobalError message="" cancelError={doNothing} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('calls cancel error on buttonclick', () => {
        const cancelErrorMock = jest.fn();
        const wrapper = shallow(
            <GlobalError message="Fahlermessage" cancelError={cancelErrorMock} />);
        expect(cancelErrorMock).toHaveBeenCalledTimes(0);
        wrapper.find(CustomButton).simulate('press');
        expect(cancelErrorMock).toHaveBeenCalledTimes(1);
    });
});
