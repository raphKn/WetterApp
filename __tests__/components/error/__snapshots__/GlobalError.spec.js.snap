// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalError renders  correctly in a snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(114,114,114,0.6)",
        "justifyContent": "center",
        "left": 0,
        "padding": 20,
        "position": "absolute",
        "top": 0,
      },
      Object {
        "height": 1334,
        "width": 750,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "paddingHorizontal": 12,
        "paddingTop": 12,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#212121",
          },
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 20,
          },
        ]
      }
    >
      Es ist ein Fehler aufgetreten
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#212121",
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      Fahlermessage
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#03A9F4",
          "height": 32,
          "justifyContent": "center",
          "margin": 20,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#212121",
            },
            Object {
              "paddingHorizontal": 16,
            },
          ]
        }
      >
        ZURÃœCK
      </Text>
    </View>
  </View>
</View>
`;

exports[`GlobalError renders no error for empty message 1`] = `null`;
